conda_build/index.py

```diff
@@ -40,6 +39,7 @@
 #  BAD BAD BAD - conda internals
 from conda.core.subdir_data import SubdirData
 from conda.models.channel import Channel
+from conda_index.index import update_index as _update_index
```

Comment:
Users have been silently updated to the new `conda-index` via this import. Have we considered deprecating this module first with a warning? What's the story in the future? Will `conda_build.index` always exist? If not, should we add a module-level warning (bis)?

Reply:
How do we add a module warning without triggering it when conda-build imports

Reply:
I think the goal would be eventually to replace this code completely with the implementation in conda-index via an own subcommand. There are two groups of users here:

- End-users of the index subcommand, they can continue to use it, and they shouldn't see any difference to the built-in index feature.
- Makers who import from `conda_build.index` in their scripts/tools and rely on the Python API. They'll need to be provided with enough information and a rollout plan to make the necessary changes.

I only worry about the second group, and how to reach them. They'll unlikely read the conda-build changelog, so we'll need to let them know via `DeprecationWarning` instead. I don't think it has to be a module-level warning, maybe a function wrapper like `conda.deprecations` would suffice.

Reply:
Updated to expose all the old API's with the old implementation, while printing warnings.